controller_manager:
  ros__parameters:
    update_rate: 100  # Hz

    joint_state_broadcaster:
      type: joint_state_broadcaster/JointStateBroadcaster

    diff_drive_controller:
      type: diff_drive_controller/DiffDriveController

diff_drive_controller:
  ros__parameters:
    left_wheel_names: ["left_wheel_joint"]
    right_wheel_names: ["right_wheel_joint"]

    wheel_separation: 0.235
    wheel_radius: 0.03

    # 运动学参数
    linear:
      x:
        has_velocity_limits: true
        max_velocity: 1.0
        min_velocity: -1.0
        has_acceleration_limits: true
        max_acceleration: 0.8
        min_acceleration: -0.8
        has_jerk_limits: false
        max_jerk: 0.0
        min_jerk: 0.0
    
    angular:
      z:
        has_velocity_limits: true
        max_velocity: 1.0
        min_velocity: -1.0
        has_acceleration_limits: true
        max_acceleration: 0.6
        min_acceleration: -0.6
        has_jerk_limits: false
        max_jerk: 0.0
        min_jerk: 0.0

    # 发布参数
    publish_rate: 50.0
    odom_frame_id: odom
    base_frame_id: base_link
    pose_covariance_diagonal: [0.001, 0.001, 0.001, 0.001, 0.001, 0.03]
    twist_covariance_diagonal: [0.001, 0.001, 0.001, 0.001, 0.001, 0.03]
    
    # 是否发布TF
    enable_odom_tf: true

    # 倒置配置
    cmd_vel_timeout: 0.5
    use_stamped_vel: false
    open_loop: true